desafio/desafio.js

```diff
@@ -1,32 +1,150 @@
+}
+
+console.log(listaDeProjetos, 'Lista de projetos vazia');
+
+adicionaProjeto(projeto1); // adiciona o projeto1 a lista de projetos utilizando a função criada
+adicionaProjeto(projeto2); // adiciona o projeto2 a lista de projetos utilizando a função criada
+adicionaProjeto(projeto3); // adiciona o projeto2 a lista de projetos utilizando a função criada
+
+console.log(listaDeProjetos, 'Lista de projetos com os projetos adicionados');
+
+// Exercicio 4
+
+function listarProjetos() {
+  // return `Lista de projetos: ${JSON.stringify(listaDeProjetos)}`;
+  return listaDeProjetos;
+}
+
+console.log(listarProjetos());
+
+// Exercicio 5
+
+function posicaoDoProjeto(posicao) {
+  if (listaDeProjetos[posicao] === undefined) {
+    return `Projeto não encontrado`;
+  } else {
+    return `O "${listaDeProjetos[posicao].tituloDoProjeto}" está na posição ${posicao} e seu status é "${listaDeProjetos[posicao].statusDoProjeto}"`;
+  }
+}
+
+console.log(posicaoDoProjeto(0)); // retorna projeto1
+console.log(posicaoDoProjeto(2)); // retorna projeto3
+console.log(posicaoDoProjeto(999)); // retorna Projeto não encontrado
+
+// Exercicio 6
+
+function totalDeProjetos() {
+  let objetos = listaDeProjetos.length;
+  let plural = 'projetos cadastrados';
+  let singular = 'projeto cadastrado';
+
+  if (objetos > 1) {
+    return `Temos o total de ${objetos} ${plural}`;
+  } else {
+    return `Temos o total de ${objetos} ${singular}`;
+  }
+}
+
+console.log(totalDeProjetos()); // retorna 3
+
+//Exercicio 7
+
+function buscarProjeto(idProjeto) {
+
+    // atribuir a variavel projetoEncontrado o retorno do filtro
+    // projeto é cada posição do array que o filtro vai acessar para buscar nossa condição
+    // return é o resultado da nossa condicao
+
+    // tambem poderiamos usar o metodo find
+
+    /*let projetoEncontrado = listaDeProjetos.filter(function (projeto){
+        return projeto.idProjeto === idProjeto;
+    });*/
+    let projetoEncontrado = listaDeProjetos.find(function (projeto){
+        return projeto.idProjeto === idProjeto;
+    });
+
+    if(projetoEncontrado !== undefined){
+    return projetoEncontrado;
+    } else {
+        return `Projeto nao encontrado`;
+    }
 }
 
-console.log(listaDeProjetos, 'Lista de projetos vazia')
 
-adicionaProjeto(projeto1) // adiciona o projeto1 a lista de projetos utilizando a função criada
 
-console.log(listaDeProjetos, 'Lista de projetos com um projeto')
+console.log(buscarProjeto(3))
+console.log(buscarProjeto(999))
 
+// Exercicio 8
 
+function mostrarResponsavel(idProjeto, idPessoa) {
+    let projeto = buscarProjeto(idProjeto);
+    let responsavel = projeto.pessoasResponsaveis.find(function (responsavel) {
+      return responsavel.idPessoa === idPessoa;
+    });
+    if (responsavel !== undefined){
+        return true;
+    } else {
+        return false;
+    }
+    
+  }
+  
+  console.log(mostrarResponsavel(2, 589)); // retornar true
+  console.log(mostrarResponsavel(2, 111)); // retornar false
```

Comment:
Cadê o restante? 👀

Reply:
Yasmin eu estou refazendo todo o meu código, é possível fazer um novo pull?

Reply:
Tive algumas dificuldades, mas depois de rever a aulas consegui compreender e me coloquei o desafio pessoal de reestruturar todo o código, mas eu sei que a entrega era ate sexta queria apenas que se não for atrapalhar que de uma olhada no próximo código.

Reply:
Oi! Claro que pode, sem problemas! Se precisar de apoio, me pinga no slack!